pymc/gp/hsgp_approx.py

```diff
@@ -85,6 +89,82 @@ def calc_basis_periodic(
     return phi_cos, phi_sin
 
 
+class HSGPParams(NamedTuple):
+    m: int
+    c: float
+    S: float
+
+
+def approx_hsgp_hyperparams(
+    x: np.ndarray, lengthscale_range: list[float], cov_func: str
+) -> HSGPParams:
+    """Utility function that uses heuristics to recommend minimum `m` and `c` values,
+    based on recommendations from Ruitort-Mayol et. al.
+
+    In practice, you need to choose `c` large enough to handle the largest lengthscales,
+    and `m` large enough to accommodate the smallest lengthscales.
+
+    NB: These recommendations are based on a one-dimensional GP.
+
+    Parameters
+    ----------
+    x_range : List[float]
+        The range of the input variable on which the GP is going to be evaluated.
+        Should be a list with two elements [x_min, x_max].
+    lengthscale_range : List[float]
+        The range of the lengthscales. Should be a list with two elements [lengthscale_min, lengthscale_max].
+    cov_func : str
+        The covariance function to use. Supported options are "expquad", "matern52", and "matern32".
+
+    Returns
+    -------
+    HSGPParams
+        A named tuple containing the recommended values for `m`, `c`, and `S`.
+        - `m` : int
+            Number of basis vectors. Increasing it helps approximate smaller lengthscales, but increases computational cost.
+        - `c` : float
+            Scaling factor such that L = c * S, where L is the boundary of the approximation.
+            Increasing it helps approximate larger lengthscales, but may require increasing m.
+        - `S` : float
+            The value of `S`, which is half the range of `x`.
+
+    Raises
+    ------
+    ValueError
+        If either `x_range` or `lengthscale_range` is not in the correct order.
+
+    References
+    ----------
+    - Ruitort-Mayol, G., Anderson, M., Solin, A., Vehtari, A. (2022).
+    Practical Hilbert Space Approximate Bayesian Gaussian Processes for Probabilistic Programming
+    """
+    if lengthscale_range[0] >= lengthscale_range[1]:
+        raise ValueError("One of the boundaries out of order")
+
+    X_center = (np.max(x, axis=0) - np.min(x, axis=0)) / 2
+    Xs = x - X_center
+    S = np.max(np.abs(Xs), axis=0)
```

Comment:
After talking to @bwengals , I changed the computation here, to make sure it matches the definition of `S` in `set_boundary`.
The previous version was only doing;
```python
S = (x_range[1] - x_range[0]) / 2
```
If that looks good, I think we're good to merge 🥳

Reply:
yah good change.  Hit the merge button!